internal/services/compute/virtual_machine_scale_set_data_source.go

```diff
@@ -275,29 +296,53 @@ func getVirtualMachineScaleSetVMConnectionInfo(ctx context.Context, networkInter
 	}, nil
 }
 
-func flattenVirtualMachineScaleSetVM(input virtualmachinescalesetvms.VirtualMachineScaleSetVM, connectionInfo *connectionInfo) map[string]interface{} {
+func flattenVirtualMachineScaleSetVM(input virtualmachinescalesetvms.VirtualMachineScaleSetVM, connectionInfo *connectionInfo, vm *virtualmachines.VirtualMachine, mode string) map[string]interface{} {
 	output := make(map[string]interface{})
 	output["name"] = *input.Name
 	output["instance_id"] = *input.InstanceId
 
-	if props := input.Properties; props != nil {
-		if props.LatestModelApplied != nil {
-			output["latest_model_applied"] = *props.LatestModelApplied
-		}
+	if mode == "Flexible" {
+		if props := vm.Properties; props != nil {
```

Comment:
We should nil check `vm` here as well.

Reply:
Added nil check to `if mode == "Flexible"`